golem/resources/debit_note/debit_note.py

```diff
@@ -53,17 +53,19 @@ async def get_previous_debit_note(self) -> Optional["DebitNote"]:
             default=None,
         )
 
-    async def get_previous_payable_debit_note(self) -> Optional["DebitNote"]:
-        """Get previous payable debit note."""
-        return max(
+    async def get_previous_debit_notes_count(self) -> int:
+        """Get previous debit notes count."""
+        return len([dn for dn in self.activity.debit_notes if dn.created_at < self.created_at])
```

Comment:
Instead of list comprehension, just provide a plain generator.
```suggestion
        return len(dn for dn in self.activity.debit_notes if dn.created_at < self.created_at)
```

Reply:
len dosen't work with generators